src/Utilities/ErrorHandling/AbortWithErrorMessage.cpp

```diff
@@ -3,70 +3,18 @@
 
 #include "Utilities/ErrorHandling/AbortWithErrorMessage.hpp"
```

Comment:
Choosing a random line for threading purposes.

I tested this out locally, and while it looks good when it works, it fails really badly when it doesn't.  Here's part of the output from the `Unit.ErrorHandling.FloatingPointExceptions.DivByZero` test:
```
  0. [error handling]
    1. [error handling]
    2. (anonymous namespace)::fpe_signal_handler(int) in /home/wthrowe/spectre/
  src/Utilities/ErrorHandling/FloatingPointExceptions.cpp:29
    3. 
    4. C_A_T_C_H_T_E_S_T_17() in /home/wthrowe/spectre/tests/Unit/
  ErrorHandling/Test_FloatingPointExceptions.cpp:64
    5. Catch::TestInvokerAsFunction::invoke() const in /usr/include/catch2/
  catch.hpp:14329
    6. Catch::TestCase::invoke() const in /usr/include/catch2/catch.hpp:14168
```
This is mostly very nice, although there is a completely blank stack frame.

Without libbacktrace available the output is nearly useless:
```
  0. 
    1. 
    2. 
    3. 
    4. 
    5. Catch::TestInvokerAsFunction::invoke() const
    6. Catch::TestCase::invoke() const
```

The current version is not pretty, but it does give some hint as to what is going on and the addr2line information is there to get the file and line number:
```
/home/wthrowe/spectre-work/lib/libErrorHandling.so(+0x5025) [0x7fc8a6a9f025] Address for addr2line: 0x5025
/lib64/libc.so.6(+0x3d7a0) [0x7fc8a532f7a0] Address for addr2line: 0x3d7a0
/home/wthrowe/spectre-work/lib/libTest_ErrorHandling.so(+0xbba7) [0x7fc8cf846ba7] Address for addr2line: 0xbba7
bin/RunTests() [0x48b673] Address for addr2line: 0x48b673
bin/RunTests() [0x485b34] Address for addr2line: 0x485b34
```

If we're going with this, I think we need to make libbacktrace (either the standalone one or the gcc internal one) a hard requirement

Reply:
You should be getting useful output without libbacktrace. It's only needed to get line numbers. For instance on my macOS machine I don't have it, but I get a full backtrace with demangled symbols. The necessary requirement for that is only the `-rdynamic` linker flag, so we should check what's going wrong there on your system. See docs: https://www.boost.org/doc/libs/1_78_0/doc/html/stacktrace/configuration_and_build.html

Reply:
Not sure what you want me to look at there.  I've checked that `-rdynamic` is being passed.  Have you tried it on a Linux machine?

Reply:
I tried on Minerva and Wheeler. Neither has libbacktrace available.

I pushed another fixup that falls back to another possible mode in Boost, using addr2line. Please give it a try.

Also, I think it would be more useful to test with a piece of code that calls into our code base, not just unit-test internals. For instance, I added an assert somewhere in the Domain constructor.

Reply:
No improvement, and it now takes over a minute to run.

Putting an error in the `Domain` constructor is not much different:
```
  0. 
    1. 
    2. Domain<1ul>::Domain(std::vector<Block<1ul>, std::allocator<Block<1ul> >
  >)
    3. 
    4. 
    5. Catch::TestInvokerAsFunction::invoke() const in /usr/include/catch2/
  catch.hpp:14329
    6. Catch::TestCase::invoke() const in /usr/include/catch2/catch.hpp:14168
```

Making sure the FPE case works is actually more important, because for an ERROR you at least get the function and line number where the macro was called.

Reply:
Ok if I remember correctly this worked much better on Wheeler. Maybe try with `-fomit-frame-pointer`? I'm not sure how I can help you debug this without access to your system, perhaps we can get in a call some time

Reply:
This now falls back to the non-pretty stack frames when demangling fails, and prints addr2line addresses.